lib/ordered_set.dart

```diff
@@ -72,19 +72,7 @@ class OrderedSet<E> extends IterableMixin<E> implements Iterable<E> {
 
   /// Remove all elements that match the [test] condition, returns the removed elements
   Iterable<E> removeWhere(bool Function(E element) test) {
-    List<E> _removed = [];
-    for (final es in _backingSet.toList()) {
-      final removed = es.where(test);
-      _removed.addAll(removed);
-      _length -= removed.length;
-
-      if (removed.length == es.length) {
-        _backingSet.remove(es);
-      } else {
-        es.removeWhere(test);
-      }
-    }
-    return _removed;
+    return this.where(test).toList()..forEach((e) {this.remove(e);});
```

Comment:
we can use an inline lambda here

```suggestion
    return this.where(test).toList()..forEach((e) => this.remove(e));
```

Reply:
Nice catch. How about converting it to oneliner then?:)